### PR TITLE
fix(build-plguin-rax-app): error in dev and build mode

### PR DESCRIPTION
**Pull request by ChrisCindy**
- fix the bug that you can't build bundles when only using miniapp
- fix the bug that log results is mixed up
- fix the bug that in dev mode log can't be printed correctly.
- fix the bug that kraken target can't get config correctly.

### CODE REVIEW

File: packages/build-plugin-rax-app/src/build.js [hunk 1]
```diff
@@ -21,78 +40,106 @@ module.exports = ({ registerTask, context, onHook }, options = {}) => {
...
-            break;
-        }
-        onHook('after.build.compile', async() => {
-          const mpInfo = await jsx2mpBuilder(context, config);
-          if (mpInfo.err || mpInfo.stats.hasErrors()) {
-            mpBuildErr = mpInfo;
-          }
+      } else if (buildScriptsDevTargets.length) {
+        onHook('after.build.compile', async () => {
+          mpBuildErr = await invokeJSX2MPBuilder(context, options[target]);
```
Comment (SoloJiang) [inline packages/build-plugin-rax-app/src/build.js]: `mpBuild` 相关的命名改下吧

File: packages/build-plugin-rax-app/src/build.js [hunk 2]
```diff
@@ -21,78 +40,106 @@ module.exports = ({ registerTask, context, onHook }, options = {}) => {
...
-  });
-};
+/**
+ *
+ * @param {*} context
+ * @param {*} config
+ */
+async function invokeJSX2MPBuilder(context, config) {
+  const jsx2mpBuilder = require('./config/miniapp/compile/build');
+  const mpInfo = await jsx2mpBuilder(context, config);
```
Comment (SoloJiang) [inline packages/build-plugin-rax-app/src/build.js]: mpInfo +1

**Comment by yacheng** [comment]
本地测试小程序运行时方案，start 和 build 命令无异常

**Comment by yacheng** [comment]
build-plguin-rax-app 近期改动较多且均未正式发布，建议升级一个 beta 版本先提供业务试用

File: packages/build-plugin-rax-app/src/build.js [hunk 3]
```diff
@@ -21,78 +40,106 @@ module.exports = ({ registerTask, context, onHook }, options = {}) => {
...
-        }
+      } else if (buildScriptsBuildTargets.length) {
         onHook('after.build.compile', async() => {
-          const mpInfo = await jsx2mpBuilder(context, config);
-          if (mpInfo.err || mpInfo.stats.hasErrors()) {
-            mpBuildErr = mpInfo;
-          }
+          jsx2mpBuildErr = await invokeJSX2MPBuilder(context, options[target]);
         });
+      } else if (jsx2mpBuildTargets.length) {
```
Comment (wssgcg1213) [inline packages/build-plugin-rax-app/src/build.js]: 尽量写 `> 0`  布尔值

**Pull request merged**